lib/src/main/java/com/what3words/components/maps/wrappers/W3WMapBoxWrapper.kt

```diff
@@ -807,40 +837,96 @@ class W3WMapBoxWrapper(
+                bitmapFromDrawableRes(context, markerFillColor)?.let { bitmap ->
+                    mapView.getStyle { style ->
+                        style.addImage(
+                            String.format(PIN_ID_PREFIX, data.markerColor.toString()),
+                            bitmap
                         )
-                    }
-                    if (!style.styleLayerExists(data.suggestion.coordinates.generateUniqueId().toString())) {
-                        style.addLayer(
-                            symbolLayer(data.suggestion.coordinates.generateUniqueId().toString(), data.suggestion.words) {
-                                iconImage(
-                                    String.format(
-                                        CIRCLE_ID_PREFIX,
-                                        data.markerColor.toString()
+                        if (!style.styleSourceExists(data.address.words)) {
+                            style.addSource(
+                                geoJsonSource(data.address.words) {
+                                    geometry(
+                                        Point.fromLngLat(
+                                            center.lng,
+                                            center.lat
+                                        )
                                     )
-                                )
-                                iconAnchor(IconAnchor.CENTER)
-                                iconAllowOverlap(true)
-                            }
+                                }
+                            )
+                        }
+                        if (!style.styleLayerExists(
+                                center.generateUniqueId().toString()
+                            )
+                        ) {
+                            style.addLayer(
+                                symbolLayer(
+                                    center.generateUniqueId().toString(),
+                                    data.address.words
+                                ) {
+                                    iconImage(
+                                        String.format(
+                                            PIN_ID_PREFIX,
+                                            data.markerColor.toString()
+                                        )
+                                    )
+                                    iconAnchor(IconAnchor.BOTTOM)
+                                    iconAllowOverlap(true)
+                                }
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /** [drawCircle] will be responsible for the drawing of the w3w circle pins (i.e [R.drawable.ic_marker_circle_red]) by adding them to [MapboxMap.style] using a [SymbolLayer].*/
+    private fun drawCircle(data: W3WAddressWithStyle) {
+        data.address.center?.let { center ->
```

Comment:
Maybe I'm wrong but in W3WGoogleMapWrapper, we have a function to draw the pin at the word when the coordinates are missing, but I don't see a similar function here.